internal/cluster/validations/validations.go

```diff
@@ -420,6 +420,33 @@ func ValidateIPAddresses(ipV6Supported bool, obj interface{}) error {
 	if err != nil {
 		return err
 	}
+
+	switch c := obj.(type) {
+	case *models.ClusterCreateParams:
+		targetConfiguration := common.Cluster{}
+		targetConfiguration.ClusterNetworks = c.ClusterNetworks
+		targetConfiguration.ServiceNetworks = c.ServiceNetworks
+		targetConfiguration.MachineNetworks = c.MachineNetworks
+
+		api := ""
+		if c.APIVip != nil {
+			api = *c.APIVip
+		}
+
+		dualStack := network.CheckIfClusterIsDualStack(&targetConfiguration)
+		if dualStack {
+			err = network.VerifyVips(nil, string(c.MachineNetworks[0].Cidr), api, c.IngressVip, true, nil)
+			if err != nil {
+				return common.NewApiError(http.StatusBadRequest, err)
+			}
+		} else {
+			if api != "" {
+				err := errors.Errorf("Providing API VIP during cluster creation is allowed only for dual-stack clusters.")
```

Comment:
Why only dual stack ?

Reply:
My motivations is as follows - currently for single stack there is no logic of providing API VIP during cluster creation. Till now the field was transparently dropped. As I'm adding a scenario for a use case that has been reported, I need to have a flow of "dual-stack + API VIP".

Looks like no one ever complained that they cannot provide API VIP when creating a single-stack cluster. This means that
* either this flow is not supported
* or no one ever used it

Depending on the answer, I may want to keep the error here or again transparently drop the value. What I definitely don't want to do in this PR is to revise all the possible single-stack flows and decide when API VIP may or may not be accepted (because again, today the API VIP is always dropped).

Any thoughts?

Reply:
I think for simplicity single and dual stack should behave similarly, except that dual stack has several networks and not one.

Reply:
Correct. I have added more flows there to not just blindly forbid API VIP for single-stack. At the same time, for single-stack we have a lot of logic around that does e.g. machine network autocalculation and this affects VIPs, so I'd appreciate a review on that here.

I tried to add unit-tests for the most common combinations I could invent. I guess the most "error-prone" scenario will be single-stack where user provides API VIP and at the same time relies on the autocalculation of machine networks.

At the same time for the single-stack we may have a difference in how we treat SNO and multi-node, so yeah, it's all a bit complex.